src/uu/dd/src/dd.rs

```diff
@@ -0,0 +1,1185 @@
+                wstat += wstat_update;
+            }
+        };
+        // Update Prog
+        prog_tx.send(ProgUpdate {
+            read_stat: rstat,
+            write_stat: wstat,
+            duration: start.elapsed(),
+        })?;
+    }
+
+    if o.cflags.fsync {
+        o.fsync()?;
+    } else if o.cflags.fdatasync {
+        o.fdatasync()?;
+    }
+
+    match i.print_level {
+        Some(StatusLevel::Noxfer) | Some(StatusLevel::None) => {}
+        _ => print_transfer_stats(&ProgUpdate {
+            read_stat: rstat,
+            write_stat: wstat,
+            duration: start.elapsed(),
+        }),
+    }
+    Ok(())
+}
+
+/// Perform the copy/convert operations. File backed output version
+/// Note: The body of this function should be kept identical to dd_stdout. This is definitely a problem from a maintenance perspective
+/// and should be addressed (TODO). The problem exists because some of dd's functionality depends on whether the output is a file or stdout.
+fn dd_fileout<R: Read>(mut i: Input<R>, mut o: Output<File>) -> Result<(), Box<dyn Error>> {
+    let mut rstat = ReadStat {
+        reads_complete: 0,
+        reads_partial: 0,
+        records_truncated: 0,
+    };
+    let mut wstat = WriteStat {
+        writes_complete: 0,
+        writes_partial: 0,
+        bytes_total: 0,
+    };
+    let start = time::Instant::now();
+    let bsize = calc_bsize(i.ibs, o.obs);
+
+    let prog_tx = {
+        let (tx, rx) = mpsc::channel();
+        thread::spawn(gen_prog_updater(rx, i.print_level));
+        tx
+    };
+
+    while below_count_limit(&i.count, &rstat, &wstat) {
+        // Read/Write
+        let loop_bsize = calc_loop_bsize(&i.count, &rstat, &wstat, i.ibs, bsize);
+        match read_helper(&mut i, loop_bsize)? {
+            (
+                ReadStat {
+                    reads_complete: 0,
+                    reads_partial: 0,
+                    ..
+                },
+                _,
+            ) => break,
+            (rstat_update, buf) => {
+                let wstat_update = o.write_blocks(buf)?;
+
+                rstat += rstat_update;
+                wstat += wstat_update;
+            }
+        };
+        // Update Prog
+        prog_tx.send(ProgUpdate {
+            read_stat: rstat,
+            write_stat: wstat,
+            duration: start.elapsed(),
+        })?;
+    }
+
+    if o.cflags.fsync {
+        o.fsync()?;
+    } else if o.cflags.fdatasync {
+        o.fdatasync()?;
+    }
+
+    match i.print_level {
+        Some(StatusLevel::Noxfer) | Some(StatusLevel::None) => {}
+        _ => print_transfer_stats(&ProgUpdate {
+            read_stat: rstat,
+            write_stat: wstat,
+            duration: start.elapsed(),
+        }),
+    }
+    Ok(())
+}
+
+fn append_dashes_if_not_present(mut acc: Vec<String>, mut s: String) -> Vec<String> {
+    if !s.starts_with("--") && !s.starts_with('-') {
+        s.insert_str(0, "--");
+    }
+    acc.push(s);
+    acc
+}
```

Comment:
So if I'm understanding this correctly this is to work around the fact that `dd` does not take arguments in the `--option` form but as `option=` (without the dashes). However this leads to some problems: Imagine invoking `dd` like this: `dd if foo.file`. This will get translated to `dd --if --foo.file`. This will result in an error, but the error will say that `Found argument '--foo.file' which wasn't expected`. This is not the right error message.